R/use_extendr.R

```diff
@@ -37,9 +37,32 @@ use_extendr <- function(path = ".",
 
   rlang::check_installed("usethis")
 
-  rextendr_setup(path = path)
+  # Root path computed from user input
+  root_path <- try_get_root_path(path)
+  # Root path computed from `{usethis}`
+  usethis_proj_path <- try_get_proj_path()
+
+  # If they do not match, something is off, try to set up temporary project
```

Comment:
what? what's going on here?

Reply:
So the code that we run depends on two things - path that we get from the user (impacts direct writes or rprojroot) and active usethis project (impacts all calls to usethis). If one of these two is out of whack, something will not work. Now in the easiest case of creating new package and immediately running use_extendr, it always works. But if your run use_extendr later or from a different folder, we need to adjust for that. We do this with temporary path and project, pretending we are in a correct folder to execute everything, and then we restore the original project/path configuration.

Reply:
seems odd, but to be fair, I have alerted to similar issues elsewhere in the code. As always, my policy is to merge what works reliable, and improve later on. godspeed @Ilia-Kosenkov !